placement/algo/mirrored.go

```diff
@@ -197,16 +206,25 @@ func (a mirroredAlgorithm) ReplaceInstances(
 	leavingInstanceIDs []string,
 	addingInstances []placement.Instance,
 ) (placement.Placement, error) {
-	if err := a.IsCompatibleWith(p); err != nil {
+	err := a.IsCompatibleWith(p)
+	if err != nil {
 		return nil, err
 	}
 
 	if len(addingInstances) != len(leavingInstanceIDs) {
 		return nil, fmt.Errorf("could not replace %d instances with %d instances for mirrored replace", len(leavingInstanceIDs), len(addingInstances))
 	}
 
-	p, err := placement.MarkAllShardsAsAvailable(p)
-	if err != nil {
+	var (
+		nowNanos               = a.opts.NowFn()().UnixNano()
+		shouldCheckCutoverTime = true
+	)
+	if allLeaving(p, addingInstances, nowNanos) && allInitializing(p, leavingInstanceIDs, nowNanos) {
```

Comment:
This seems backwards to me, how come we don't want to check if the leaving instances are leaving and the added instances are initializing? i.e

`if allLeaving(p, leavingInstanceIDs, nowNanos) && allInitializing(p, addingInstances, nowNanos) {`

Reply:
this is exactly what we want, we often need to add instances that are leaving in the placement, for example you have 10 instances in normal (Available) state, and you removed 2 of them, the 2 instances will become "leaving" in the placement, then it's totally ok if we want to add the 2 "leaving" instances back to the placement